autogpt_platform/backend/backend/data/notifications.py

```diff
@@ -0,0 +1,333 @@
+import logging
+from datetime import datetime, timedelta
+from enum import Enum
+from typing import Annotated, Generic, Optional, Type, TypeVar, Union
+
+from prisma import Json
+from prisma.enums import NotificationType
+from prisma.models import NotificationEvent, UserNotificationBatch
+from prisma.types import UserNotificationBatchWhereInput
+
+# from backend.notifications.models import NotificationEvent
+from pydantic import BaseModel, EmailStr, Field, field_validator
+
+from .db import transaction
+
+logger = logging.getLogger(__name__)
+
+
+T_co = TypeVar("T_co", bound="BaseNotificationData", covariant=True)
```

Comment:
- `T` is usually unbound; bound type vars can get their own acronyms for clarity
- `BaseNotificationData` == `BaseModel`, so why not use `BaseModel`?

Reply:
I want only things that inherit from `BaseNotificationData`

I can fix this in a follow up since I'll probs need a contravariant too